### PR TITLE
add kwarg to disable checking number of equations and states agree

### PR DESCRIPTION
**Pull request by isaacsas**
CC @YingboMa

### CODE REVIEW

File: src/systems/abstractsystem.jl [hunk 1]
```diff
@@ -665,15 +665,17 @@ AbstractTrees.children(sys::ModelingToolkit.AbstractSystem) = ModelingToolkit.ge
 AbstractTrees.printnode(io::IO, sys::ModelingToolkit.AbstractSystem) = print(io, nameof(sys))
 AbstractTrees.nodetype(::ModelingToolkit.AbstractSystem) = ModelingToolkit.AbstractSystem
 
-function check_eqs_u0(eqs, dvs, u0)
+function check_eqs_u0(eqs, dvs, u0; check_length=true, kwargs...)
```
Comment (YingboMa) [inline src/systems/abstractsystem.jl:668]: Cannot we just do `check_length && return` in the first line of this function?

Comment (isaacsas) [reply src/systems/abstractsystem.jl:668]: Seemed like it still makes sense to check the length of `u0` is the length of the `states`, or are there cases where those could disagree?

Comment (YingboMa) [reply src/systems/abstractsystem.jl:668]: Ah, cool.

**Comment by isaacsas** [comment]
@YingboMa ok to make a new release? (Or can you?) I'll then update ReactionNetworkImporters.

**Comment by YingboMa** [comment]
Thanks! I just registered a new release.

**Comment by isaacsas** [comment]
Great, thanks!

**Pull request merged**